theano/tensor/nnet/conv.py

```diff
@@ -363,10 +365,13 @@ def getOutputShape(inshp, kshp, stride=(1, 1), mode='valid'):
         # The formula would be ceil((i + s * k - s * 1) / float(d)),
         # with s=1 for mode=='full' and s=-1 for mode=='valid'.
         # To support symbolic shapes, we express this with integer arithmetics.
-        return tuple(None if i is None or k is None
-                     else ((i - k) // d + 1) if mode == 'valid'
-                     else ((i + k + d - 2) // d)
-                     for i, k, d in zip(inshp, kshp, stride))
+        warnings.warn("The method `getOutputShape` is deprecated use"
+                      "`get_conv_output_shape` instead.")
+        return get_conv_output_shape(
+            image_shape=(None, None, inshp[0], inshp[1]),
+            kernel_shape=(None, None, kshp[0], kshp[1]),
+            border_mode=mode,
+            subsample=stride)
```

Comment:
> I would favor the second one.

I'd favor the first one, otherwise you'll need to replicate the distinction between `mode` being a string and `mode` being a tuple of int here.